enhancements/cluster-logging/api-audit-log-policy.md

```diff
@@ -0,0 +1,204 @@
+Service Delivery uses  this configuration internally for managed clusters.
+
+The goals of this proposal are:
+- Make the Kubernetes audit policy available for customer log forwarding.
+- Allow customers and Service Delivery to configure _independent_ policies and destinations for audit logs.
+
+[k8s-auditing]: https://kubernetes.io/docs/tasks/debug/debug-cluster/audit
+
+## Motivation
+
+### User Stories
+
+#### As a cluster owner, I want to store and/or forward selected API audit logs to my chosen destinations
+
+- configure an event logging policy to match my security auditing needs.
+- store and/or forward the resulting events in any destination supported by the `ClusterLogForwarder`.
+
+#### As a cluster manager, I want selected audit logs to be forwarded to my preferred log store
+
+- store/forward audit logs for internal use according to cluster manager policies, to my preferred log store.
+- independent of the users desired policy and destinations.
+
+### Goals
+
+#### Fine grained control
+
+The [Kubernetes Policy][k8s-auditing] provides detailed tuning, and has been well tested by the Kubernetes community and Service Delivery.
+
+
+#### Multiple separate audit policies
+
+For managed clusters, the cluster manager and cluster owner may have different auditing needs.
+third party cluster-auditing tools may introduce yet other auditing needs.
+
+To support _multiple_ independent audit log streams, we cannot configure filtering in the API server directly.
+Instead the API server will write complete logs (or some super-set of all desired logs)
+and the `ClusterLogForwarder` will apply a separate policy filter for each consumer.
+
+### Non-Goals
+
+- No help with semantics of audit events or design of audit policies.
+- No Hypershift support in this proposal -  [see separate hypershift proposal](hypershift-audit-logs.md)
+
+## Proposal
+
+### Workflow Description
+
+Cluster admin will
+1. Create a `Policy.audit.k8s.io` resource with the desired policy.
+2. Create an `audit` input in the `ClusterLogForwarder` linked to the policy.
+3. Create a `ClusterLogForwarder.pipeline` as normal to forward the resulting logs.
+
+### API Extensions
+
+#### New resource `Policy.audit.k8s.io`
+
+See [Kubernetes Documentation][k8s-auditing] for an example.
+
+#### Extensions to `ClusterLogForwarder.logging.openshift.io`
+
+Extended the `ClusterLogForwarder` input selector to include an `audit` section with a policy.
+For example:
+
+``` yaml
+  inputs:
+    - name: myAuditLogs
+      audit:
+        apiAuditPolicyRef: myAuditPolicy # Name of a Policy.audit.k8s.io object in same namespace.
+  outputs:
+    - name: myAuditLogSore
+      type: any supported log output ...
+  pipelines:
+    - inputRefs: [myAuditLogs]
+      outputRefs: [myAuditLogStore]
+```
+
+### Implementation Details
+
+There is already a go executable that does everything we need: 
+https://gitlab.cee.redhat.com/service/splunk-audit-exporter
```

Comment:
The audit-exporter supports more than just applying the policy - so exposing these features should either explicitly be not part of this enhancement or there must be a way these can be specified: e.g.

```
     --allowed-keys strings                 allowed annotations (default [authorization.k8s.io/decision,authorization.k8s.io/reason,managed.openshift.io/rate-limited])
      --annotations stringToString           annotations to be added to each event (--annotations=my-annotation=my-value) (default [])
      [...]
      --dedupe                               deduplicate repeated update requests (default true)
      --dedupe-cache-ttl duration            max ttl for dedupe write cache (default 48h0m0s)
      --filter-status-codes int32Slice       ignore events with these http response codes (default [404,409,422,429])
      [...]
      --rate-limit                           enable rate limiting (default true)
      --rate-limit-burst int                 per-bucket burst (default 5)
      --rate-limit-count int                 rate limit count per period (default 1)
      --rate-limit-period duration           rate limit period (default 15m0s)
```